libs/community/langchain_community/document_loaders/firecrawl.py

```diff
@@ -64,13 +63,135 @@ class FireCrawlLoader(BaseLoader):
+
+                if 'userPrompt' in params['extractorOptions']:
+                    if params['extractorOptions']['userPrompt']:
+                        params['prompt'] = params['extractorOptions']['userPrompt']
+
+                del params['extractorOptions']
+
+        scrape_keys = ['includeMarkdown', 'includeHtml', 'includeRawHtml', 'includeExtract', 'includeLinks', 'screenshot', 'fullPageScreenshot', 'onlyIncludeTags', 'removeTags']
+        for key in scrape_keys:
+            if params.get(key):
+                use_legacy_options = True
+                break
+
+        if use_legacy_options:
+            warnings.warn("Some of the used parameters are going to be deprecated in the next versions. Please check the FireCrawl documentation about v1 for updated usage.", DeprecationWarning)
+            if 'includeMarkdown' in params:
+                if params['includeMarkdown'] is False:
+                    formats.remove("markdown")
+                del params['includeMarkdown']
+
+            if 'includeHtml' in params:
+                if params['includeHtml'] is True:
+                    formats.append("html")
+                del params['includeHtml']
+
+            if 'includeRawHtml' in params:
+                if params['includeRawHtml'] is True:
+                    formats.append("rawHtml")
+                del params['includeRawHtml']
+
+            if 'includeExtract' in params:
+                if params['includeExtract'] is True:
+                    formats.append("extract")
+                del params['includeExtract']
+
+            if 'includeLinks' in params:
+                if params['includeLinks'] is True:
+                    formats.append("links")
+                del params['includeLinks']
+
+            if 'screenshot' in params:
+                if params['screenshot'] is True:
+                    formats.append("screenshot")
+                del params['screenshot']
+
+            if 'fullPageScreenshot' in params:
+                if params['fullPageScreenshot'] is True:
+                    formats.append("screenshot@fullPage")
+                del params['fullPageScreenshot']
+
+            if 'onlyIncludeTags' in params:
+                if params['onlyIncludeTags'] is True:
+                    params['includeTags'] = params['onlyIncludeTags']
+                del params['onlyIncludeTags']
+
+            if 'removeTags' in params:
+                if params['removeTags'] is True:
+                    params['excludeTags'] = params['removeTags']
+                del params['removeTags']
+
+        if 'formats' not in params:
+            params['formats'] = formats
+        print(params)
+        
+        return params
+
     def __init__(
         self,
         url: str,
         *,
         api_key: Optional[str] = None,
-        api_url: Optional[str] = None,
-        mode: Literal["crawl", "scrape"] = "crawl",
+        mode: Literal["crawl", "scrape", "map"] = "crawl",
```

Comment:
just confirming this will be a breaking change to anyone passing `api_url`. It's ok with me if it's ok on the firecrawl side!

Reply:
Oh that's not good. I just reverted that. @efriis Thank you!